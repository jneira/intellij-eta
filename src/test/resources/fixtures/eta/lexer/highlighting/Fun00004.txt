ITinline_prag ('{-#')
ITline_prag (' LANGUAGE ViewPatterns ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE TypeOperators')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE RankNTypes ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE FlexibleContexts ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITinline_prag ('{-#')
ITline_prag (' LANGUAGE UnboxedTuples ')
ITline_prag ('#-}')
WHITE_SPACE ('\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00004')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITblockComment ('-- | A record with options for explicit passing in rewrite rules.\n')
ITdata ('data')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE ('\n    ')
ITocurly ('{')
WHITE_SPACE (' ')
ITvarid ('targets')
WHITE_SPACE ('    ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITobrack ('[')
ITconid ('Target')
ITcbrack (']')
WHITE_SPACE ('\n    ')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITblockComment ('-- | Default options.\n')
ITvarid ('defaultFeldOpts')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE ('\n')
ITvarid ('defaultFeldOpts')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE (' ')
ITocurly ('{')
WHITE_SPACE (' ')
ITvarid ('targets')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITobrack ('[')
ITcbrack (']')
WHITE_SPACE (' ')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITblockComment ('-- | Insert a variable into the environment\n')
ITvarid ('localVar')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Typeable')
WHITE_SPACE (' ')
ITvarid ('b')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('VarId')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Info')
WHITE_SPACE (' ')
ITvarid ('b')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Opt')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Opt')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITvarid ('localVar')
WHITE_SPACE (' ')
ITvarid ('v')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('local')
WHITE_SPACE (' ')
ITvarsym ('$')
WHITE_SPACE (' ')
ITvarsym ('\')
ITvarid ('env')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('env')
WHITE_SPACE (' ')
ITocurly ('{')
ITvarid ('varEnv')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('v')
ITcomma (',')
WHITE_SPACE (' ')
ITconid ('SomeInfo')
WHITE_SPACE (' ')
ITvarid ('info')
ITcparen (')')
ITcolon (':')
ITvarid ('varEnv')
WHITE_SPACE (' ')
ITvarid ('env')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITblockComment ('-- | It the expression is a literal, its value is returned, otherwise 'Nothing'\n')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITforall ('forall')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
ITvarid ('dom')
WHITE_SPACE (' ')
ITvarid ('a')
ITvarsym ('.')
WHITE_SPACE (' ')
IToparen ('(')
IToparen ('(')
ITconid ('Literal')
WHITE_SPACE (' ')
ITconsym (':||')
WHITE_SPACE (' ')
ITconid ('Type')
ITcparen (')')
WHITE_SPACE (' ')
ITconsym (':<:')
WHITE_SPACE (' ')
ITvarid ('dom')
ITcparen (')')
WHITE_SPACE ('\n            ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('ASTF')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Decor')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('dom')
WHITE_SPACE (' ')
ITconsym (':||')
WHITE_SPACE (' ')
ITconid ('Typeable')
ITcparen (')')
ITcparen (')')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Maybe')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('prjF')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('C'')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Literal')
WHITE_SPACE (' ')
ITvarid ('a')
ITcparen (')')
ITcparen (')')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
ITunderscore ('_')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Nothing')
WHITE_SPACE ('\n\n')
ITvarid ('g')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToubxparen ('(#')
WHITE_SPACE (' ')
ITcubxparen ('#)')
