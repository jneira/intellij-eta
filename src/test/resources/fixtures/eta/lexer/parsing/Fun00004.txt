ITblockComment ('{-# LANGUAGE ViewPatterns #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE TypeOperators#-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE RankNTypes #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE FlexibleContexts #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE UnboxedTuples #-}')
WHITE_SPACE ('\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00004')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | A record with options for explicit passing in rewrite rules.')
WHITE_SPACE ('\n')
ITvocurly ('')
ITdata ('data')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE ('\n    ')
ITocurly ('{')
WHITE_SPACE (' ')
ITvarid ('targets')
WHITE_SPACE ('    ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITobrack ('[')
ITconid ('Target')
ITcbrack (']')
WHITE_SPACE ('\n    ')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | Default options.')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('defaultFeldOpts')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('defaultFeldOpts')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE (' ')
ITocurly ('{')
WHITE_SPACE (' ')
ITvarid ('targets')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITobrack ('[')
ITcbrack (']')
WHITE_SPACE (' ')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | Insert a variable into the environment')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('localVar')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Typeable')
WHITE_SPACE (' ')
ITvarid ('b')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('VarId')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Info')
WHITE_SPACE (' ')
ITvarid ('b')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Opt')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Opt')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('localVar')
WHITE_SPACE (' ')
ITvarid ('v')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('local')
WHITE_SPACE (' ')
ITvarsym ('$')
WHITE_SPACE (' ')
ITlam ('\')
ITvarid ('env')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('env')
WHITE_SPACE (' ')
ITocurly ('{')
ITvarid ('varEnv')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('v')
ITcomma (',')
WHITE_SPACE (' ')
ITconid ('SomeInfo')
WHITE_SPACE (' ')
ITvarid ('info')
ITcparen (')')
ITcolon (':')
ITvarid ('varEnv')
WHITE_SPACE (' ')
ITvarid ('env')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | It the expression is a literal, its value is returned, otherwise 'Nothing'')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITvarid ('forall')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
ITvarid ('dom')
WHITE_SPACE (' ')
ITvarid ('a')
ITdot ('.')
WHITE_SPACE (' ')
IToparen ('(')
IToparen ('(')
ITconid ('Literal')
WHITE_SPACE (' ')
ITconsym (':||')
WHITE_SPACE (' ')
ITconid ('Type')
ITcparen (')')
WHITE_SPACE (' ')
ITconsym (':<:')
WHITE_SPACE (' ')
ITvarid ('dom')
ITcparen (')')
WHITE_SPACE ('\n            ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('ASTF')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Decor')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('dom')
WHITE_SPACE (' ')
ITconsym (':||')
WHITE_SPACE (' ')
ITconid ('Typeable')
ITcparen (')')
ITcparen (')')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Maybe')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
IToparen ('(')
ITvarid ('prjF')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('C'')
WHITE_SPACE (' ')
IToparen ('(')
ITconid ('Literal')
WHITE_SPACE (' ')
ITvarid ('a')
ITcparen (')')
ITcparen (')')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITsemi ('')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
ITunderscore ('_')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Nothing')
WHITE_SPACE ('\n\n')
ITsemi ('')
ITvarid ('g')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
ITvarsym ('#')
WHITE_SPACE (' ')
ITvarsym ('#')
ITcparen (')')