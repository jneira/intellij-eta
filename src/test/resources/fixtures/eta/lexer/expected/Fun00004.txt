ITblockComment ('{-# LANGUAGE ViewPatterns #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE TypeOperators#-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE RankNTypes #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE FlexibleContexts #-}')
WHITE_SPACE ('\n')
ITblockComment ('{-# LANGUAGE UnboxedTuples #-}')
WHITE_SPACE ('\n')
ITmodule ('module')
WHITE_SPACE (' ')
ITconid ('Fun00004')
WHITE_SPACE (' ')
ITwhere ('where')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | A record with options for explicit passing in rewrite rules.')
WHITE_SPACE ('\n')
ITvocurly ('')
WHITE_SPACE ('')
ITdata ('data')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE ('\n    ')
ITocurly ('{')
WHITE_SPACE (' ')
ITvarid ('targets')
WHITE_SPACE ('    ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITobrack ('[')
WHITE_SPACE ('')
ITconid ('Target')
WHITE_SPACE ('')
ITcbrack (']')
WHITE_SPACE ('\n    ')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | Default options.')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('defaultFeldOpts')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('defaultFeldOpts')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('FeldOpts')
WHITE_SPACE (' ')
ITocurly ('{')
WHITE_SPACE (' ')
ITvarid ('targets')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITobrack ('[')
WHITE_SPACE ('')
ITcbrack (']')
WHITE_SPACE (' ')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | Insert a variable into the environment')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('localVar')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITconid ('Typeable')
WHITE_SPACE (' ')
ITvarid ('b')
WHITE_SPACE (' ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('VarId')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Info')
WHITE_SPACE (' ')
ITvarid ('b')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Opt')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Opt')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('localVar')
WHITE_SPACE (' ')
ITvarid ('v')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITvarid ('local')
WHITE_SPACE (' ')
ITvarsym ('$')
WHITE_SPACE (' ')
ITlam ('\')
WHITE_SPACE ('')
ITvarid ('env')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITvarid ('env')
WHITE_SPACE (' ')
ITocurly ('{')
WHITE_SPACE ('')
ITvarid ('varEnv')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('v')
WHITE_SPACE ('')
ITcomma (',')
WHITE_SPACE (' ')
ITconid ('SomeInfo')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcolon (':')
WHITE_SPACE ('')
ITvarid ('varEnv')
WHITE_SPACE (' ')
ITvarid ('env')
WHITE_SPACE ('')
ITccurly ('}')
WHITE_SPACE ('\n\n')
ITlineComment ('-- | It the expression is a literal, its value is returned, otherwise 'Nothing'')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
ITdcolon ('::')
WHITE_SPACE (' ')
ITvarid ('forall')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
ITvarid ('dom')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITdot ('.')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Literal')
WHITE_SPACE (' ')
ITconsym (':||')
WHITE_SPACE (' ')
ITconid ('Type')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITconsym (':<:')
WHITE_SPACE (' ')
ITvarid ('dom')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('\n            ')
ITdarrow ('=>')
WHITE_SPACE (' ')
ITconid ('ASTF')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Decor')
WHITE_SPACE (' ')
ITvarid ('info')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('dom')
WHITE_SPACE (' ')
ITconsym (':||')
WHITE_SPACE (' ')
ITconid ('Typeable')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Maybe')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarid ('prjF')
WHITE_SPACE (' ')
ITrarrow ('->')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('C'')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITconid ('Literal')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE ('')
ITcparen (')')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Just')
WHITE_SPACE (' ')
ITvarid ('a')
WHITE_SPACE ('\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('viewLiteral')
WHITE_SPACE (' ')
ITunderscore ('_')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
ITconid ('Nothing')
WHITE_SPACE ('\n\n')
ITsemi ('')
WHITE_SPACE ('')
ITvarid ('g')
WHITE_SPACE (' ')
ITequal ('=')
WHITE_SPACE (' ')
IToparen ('(')
WHITE_SPACE ('')
ITvarsym ('#')
WHITE_SPACE (' ')
ITvarsym ('#')
WHITE_SPACE ('')
ITcparen (')')